src/main/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -517,10 +517,12 @@ public CompositeData finishLocalSampling(String samplerType, int count) throws O
         TabularDataSupport result = new TabularDataSupport(COUNTER_TYPE);
 
         JsonArray counters = tableSamplerResult.getJsonArray((samplerType.equalsIgnoreCase("reads")) ? "read" : "write");
-        long size = 0;
+        long size = count;
```

Comment:
If count==7 but counters.size() is empty, this will leave size=7 to be returned at the end. Is this fine or not?

Reply:
good catch, I'm sending a change that would cover this one